Sources/Queries/QueryRequest.swift

```diff
@@ -0,0 +1,79 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import CryptoKit
+import Foundation
+
+import Firebase
+
+/// A QueryRequest is used to get a QueryRef to a Data Connect query using the specified query name
+/// and input variables to the query
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
+struct QueryRequest<Variable: OperationVariable>: OperationRequest, Hashable, Equatable {
+  private(set) var operationName: String
+  private(set) var variables: Variable?
```

Comment:
This is minor, but I would think the _variable**s**_ would map to an array of `Variable` (e.g. `[Variable]`)?

Reply:
The fields of the container struct map to the names of the variables of that operation.